src/engine.rs

```diff
@@ -0,0 +1,187 @@
+use crate::bytes::{CharIterator, Confidence, Encoding};
+use crate::dns::ResolveType;
+use crate::html5::parser::document::{Document, DocumentBuilder, DocumentHandle};
+use crate::html5::parser::Html5Parser;
+use crate::net::errors::Error;
+use crate::net::http::headers::Headers;
+use crate::net::http::request::Request;
+use crate::net::http::response::Response;
+use crate::timing::{Timing, TimingTable};
+use crate::types::ParseError;
+use cookie::CookieJar;
+use core::fmt::Debug;
+use core::str::FromStr;
+use reqwest::header::HeaderName;
+use reqwest::Method;
+use url::Url;
+
+/// Response that is returned from the fetch function
+pub struct FetchResponse {
```

Comment:
This is the response from the gosub-engine when you fetch a url. Probably will change in the future, but I think its a start to generate data on a useragent